### DISCUSSION TITLE
libLISA - Instruction Discovery and Analysis on x86-64

### DISCUSSION THREAD

**Discussion by conversy** [ptitSeb/box64#1963] [Ideas]
FYI. I figured it could be of interest for Box86/64: [liblisa](https://liblisa.nl/)

> libLISA is a tool that can fully automatically scan instruction space, discover instructions and synthesize their semantics.

Cheers,

Stéphane

**Comment by ksco** [comment]
Since it can probe the semantics of all instructions automatically, this approach could theoretically generate a userspace x86-64 emulator with similar functionality to box64 or qemu-user, but it's still quite hard to make it a fast one.

**Reply by conversy** [reply]
Yes, AFAIU box64 is translating and optimizing in a sensible way... But liblisa might be used to check things. I'm not even sure, just wanted to let you know.

Great works btw, congratulations for what you guys have done so far...

**Reply by ksco** [reply]
For the checking, we have `BOX64_DYNAREC_TEST` to compare the microarchitecture state between DynaRec and Interpreter opcode by opcode, but yeah, would be interesting to compare against an emulator generated from real hardware!
